### ISSUE TITLE
Support Additional Virtual Table Functionality in apsw 3.41.0.0

### ISSUE THREAD

**Issue by cancan101** [betodealmeida/shillelagh#344]
See [release notes](https://github.com/rogerbinns/apsw/releases/tag/3.41.0.0), specifically:
- https://github.com/rogerbinns/apsw/issues/332
- https://github.com/rogerbinns/apsw/issues/278
- https://github.com/rogerbinns/apsw/issues/188

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by betodealmeida** [comment]
Yeah! I started working on this, but had some problems compiling from source. With the new release I should be able to jus use the wheel. Will finish the branch.

**Comment by gfieremans** [comment]
@betodealmeida - see discussion in https://github.com/rogerbinns/apsw/discussions/428. You need to increment the value of the argvIndex in your function BestIndexObject():

`        for i, constraint in enumerate(constraints_used):
            if isinstance(constraint, tuple):
                index_info.set_aConstraintUsage_argvIndex(i, constraint[0]**+1**)
                index_info.set_aConstraintUsage_omit(i, constraint[1])
`

**Comment by betodealmeida** [comment]
Ah, thanks, I'll fix it!

**Comment by gfieremans** [comment]
By the way - I applied your bestindex branch to 1.2.0 and built it and everything is working great. Also was able to get the multiselect constraintargs in the Filter() method. if you're interested I can share my changes for that - although I wonder whether that's not too specific to my use case.

**Comment by betodealmeida** [comment]
> Also was able to get the multiselect constraintargs in the Filter() method.

You mean supporting `IN` statements? That would be great!

**Comment by gfieremans** [comment]
Yes, that's why I went with the BestIndexObject you provided. Shall I checkout a branch off the bestindex branch?

**Comment by betodealmeida** [comment]
The best index branch has been merged in master, so you can just use master.

**Comment by betodealmeida** [comment]
(I should start deleting branches.)
